Sources/StreamVideoSwiftUI/CallView/ScreenSharing/ScreenSharingView.swift

```diff
@@ -73,23 +73,47 @@ public struct ScreenSharingView<Factory: ViewFactory>: View {
             contentMode: .scaleAspectFit
         ) { view in
             if let track = screenSharing.participant.screenshareTrack {
-                log.debug("adding screensharing track to a view \(view)")
+                log.info("Found \(track.kind) track:\(track.trackId) for \(screenSharing.participant.name) and will add on \(type(of: self)):\(identifier))", subsystems: .webRTC)
                 view.add(track: track)
             }
         }
-        .frame(
-            width: viewModel.hideUIElements ? videoSize.width : nil,
-            height: viewModel.hideUIElements ? videoSize.height : nil
-        )
+        .clipShape(RoundedRectangle(cornerRadius: 16))
     }
 
     private var videoSize: CGSize {
         if viewModel.hideUIElements {
-            return .init(width: availableFrame.size.height, height: availableFrame.size.width)
+            return .init(
+                width: frame.width,
+                height: frame.height - participantsStripFrame.height - innerItemSpace
+            )
+        } else {
+            return .init(
+                width: frame.width,
+                height: frame.height - innerItemSpace
+            )
+        }
+    }
+
+    private var itemsVisibleOnScreen: CGFloat {
```

Comment:
Hmm, can you explain this one, please?

Reply:
Sure, on iPhone [we have 2 items visible](https://www.figma.com/file/xAr946UfaDAOTDfX29WfEC/Video-Calling-UI-Kit-%F0%9F%9A%80?type=design&node-id=951-44563&mode=design&t=qGmSTWQB6TtgTxvF-4) while [on iPad we get 4](https://www.figma.com/file/xAr946UfaDAOTDfX29WfEC/Video-Calling-UI-Kit-%F0%9F%9A%80?type=design&node-id=1095-100718&mode=design&t=qGmSTWQB6TtgTxvF-4). The only additional thing here, is that on iPad portrait we show 3 instead of 4. That's in order to maintain the aspect ratio and the tiles in a "not too small" size.

Reply:
Still not convinced by this approach. What if we have portrait on iPhone?

Reply:
*sorry landscape